connection_helper.go

```diff
@@ -39,3 +42,12 @@ func (c *Connection) writeQuery(token int64, q []byte) error {
 
 	return c.writeData(data)
 }
+
+func (c *Connection) contextFromConnectionOpts() context.Context {
+	sum := c.opts.ReadTimeout + c.opts.WriteTimeout
```

Comment:
Ok fair enough, I am still a little worried about changing the existing behaviour but hopefully the new behaviour should not cause too many issues.